### PR TITLE
94 refactor checkbox logic

### PR DESCRIPTION
**Pull request by hellesvaasand**

### CODE REVIEW

File: sip-viewer/src/app/session-table/session-table.component.ts [hunk 1]
```diff
@@ -52,12 +55,13 @@ export class SessionTableComponent implements AfterViewInit {
...
-        this.tableData = []; // Empty the table to avoid duplicate copies of the sessions
-        this.selection.clear(); // Set all sessions as not selected
+        // Empty table and deselect sessions to avoid duplicates when uploading new file
+        this.tableData = [];
+        this.selection.clear();
 
         // Extract unique session IDs and add time, session ID, sender and receiver to lists
         const sessionIDs = new Set<string>();
-        const phoneNumberPattern = /<sip:?(\+?\d+)@/;
+        const phoneNumberPattern = /<sip:?(\+?\d+)@/; // Kunne det være annet type nummer?
```
Comment (moltubaer) [inline sip-viewer/src/app/session-table/session-table.component.ts]: Change the comment to english or remove it

File: sip-viewer/src/app/session-table/session-table.component.ts [hunk 2]
```diff
@@ -97,34 +101,33 @@ export class SessionTableComponent implements AfterViewInit {
...
-    this.dataService.updateSelectedSessionsByList(
-      this.sessionIDsToSendToDataService
-    );
-    if (this.sessionIDsToSendToDataService.length === 0) {
-      this.headerCheckbox.toggle();
-      this.selection.clear();
+    const selectedSessions = this.getSelectedSessions();
+    if (selectedSessions.includes(row)) {
+      this.selection.deselect(row);
+    } else if (!selectedSessions.includes(row)) {
```
Comment (moltubaer) [inline sip-viewer/src/app/session-table/session-table.component.ts]: Check if else if is necessary. If not, then you can just change it to else. This makes the code much more readable

**Review by samer-12 — APPROVED** [review]
LVGTM

**Review by moltubaer — APPROVED** [review]
lgtm

**Pull request merged**